### PR TITLE
feat(clients): add helper to check if an index exists

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-2775](https://algolia.atlassian.net/browse/DI-2775). closes https://github.com/algolia/algoliasearch-client-php/issues/732

Add the `indexExists` helpers which uses `getSettings` behind the hood.

[DI-2775]: https://algolia.atlassian.net/browse/DI-2775?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: scripts/cli/index.ts [hunk 1]
```diff
@@ -40,6 +40,11 @@ const flags = {
 
 program.name('cli');
 
+program.hook('preAction', () => {
+  // restore the cursor because sometime it's broken
+  process.stdout.write('\x1B[?25h');
```
Comment (shortcuts) [inline scripts/cli/index.ts:45]: ty

File: tests/output/python/pyproject.toml [hunk 2]
```diff
@@ -18,3 +18,6 @@ pytest-aiohttp = "1.0.5"
 [tool.pytest.ini_options]
 asyncio_mode = "auto"
 testpaths = ["tests"]
+
+[tool.ruff]
+unsafe-fixes = true
```
Comment (shortcuts) [inline tests/output/python/pyproject.toml:23]: D:

**Review by shortcuts — COMMENTED** [review]
looks great! waiting for the gen

ggggg

File: tests/output/python/pyproject.toml (hunk 2 above)
Comment (millotp) [reply tests/output/python/pyproject.toml:23]: this is just to convert `require _req == True` to `require __req is True`, its not happy otherwise

**Review by shortcuts — APPROVED** [review]
it looks so puuurfect

**Pull request merged**